src/snarks/groth16/mpc/phase2.tcc

```diff
@@ -0,0 +1,794 @@
+#ifndef __ZETH_SNARKS_GROTH16_MPC_PHASE2_TCC__
+#define __ZETH_SNARKS_GROTH16_MPC_PHASE2_TCC__
+
+#include "chacha_rng.hpp"
+#include "libff/common/rng.hpp"
+#include "mpc_utils.hpp"
+#include "powersoftau_utils.hpp"
+#include "util.hpp"
+
+namespace libzeth
+{
+
+template<typename ppT>
+srs_mpc_phase2_accumulator<ppT>::srs_mpc_phase2_accumulator(
+    const srs_mpc_hash_t cs_hash,
+    const libff::G1<ppT> &delta_g1,
+    const libff::G2<ppT> &delta_g2,
+    libff::G1_vector<ppT> &&H_g1,
+    libff::G1_vector<ppT> &&L_g1)
+    : delta_g1(delta_g1), delta_g2(delta_g2), H_g1(H_g1), L_g1(L_g1)
+{
+    memcpy(this->cs_hash, cs_hash, sizeof(srs_mpc_hash_t));
+}
+
+template<typename ppT>
+bool srs_mpc_phase2_accumulator<ppT>::operator==(
+    const srs_mpc_phase2_accumulator<ppT> &other) const
+{
+    return !memcmp(cs_hash, other.cs_hash, sizeof(srs_mpc_hash_t)) &&
+           (delta_g1 == other.delta_g1) && (delta_g2 == other.delta_g2) &&
+           (H_g1 == other.H_g1) && (L_g1 == other.L_g1);
+}
+
+template<typename ppT>
+bool srs_mpc_phase2_accumulator<ppT>::is_well_formed() const
+{
+    return delta_g1.is_well_formed() && delta_g2.is_well_formed() &&
+           libzeth::container_is_well_formed(H_g1) &&
```

Comment:
`libzeth::`, you're in the `libzeth` namespace, do you really need this prefix?

Reply:
removed namespace prefix